wfl/Controllers/RestaurantsController.cs

```diff
@@ -2,13 +2,16 @@
 using System.Threading.Tasks;
 using Symphono.Wfl.Models;
 using Symphono.Wfl.Database;
-
+using Microsoft.Practices.Unity;
 
 namespace Symphono.Wfl.Controllers
 {
     [RoutePrefix("api/Restaurant")]
     public class RestaurantsController : ApiController
     {
+        IUnityContainer container = new UnityContainer();
```

Comment:
We'll want to change how this container is created and registered.  I'm 99.9% sure that a new controller is created for every request.  Which means in this case we are creating a new unity container upon every request.  We then register container elements on every request.  These are all expensive operations.

On top of this, we have not actually reduced our dependencies from this class.  We still specify all of the same dependencies, plus we have added a Unity container as a new dependency.  At this point we are now less re-usable and less testable than when we started.

To solve these issues, I would expect the controller itself to take in all dependencies within the constructor, which allows a consumer to know what it depends on (in this case that means the constructor would take in `IDBManager` and only `IDBManager`).  That would remove all explicit dependencies, and also start reducing the responsibilities of this controller to simply acting as a controller, versus understanding all of our inversion of control.

We then would need to find a place which runs at application startup (ASP.NET as startup hooks for configuration).  We can then create our unity container there, and then provide that as the container to ASP.NET so that it will use our configured container to create all controllers (which is how we will have our dependencies provided to us).  It will then deal with ensuring all of our dependencies also have their dependencies satisfied as well.